lesson 1/README.md

```diff
@@ -2,6 +2,93 @@
 
 ##  Github
 
+### Ideas behind a perfect repo
+
+#### 1. What should the master branch contain?
+Working and well-tested code that has been reviewed by peers.
+
+#### 2. Non-master branches?
+    ..1. Each of these branch should mostly have one (maybe two if you are pair-programming a feature) person working on it. 
+    ..2. Each branch should add a significant working change to the master. Usually each branch adds a working feature to the master
+
+#### 3. Issues:
+    ..1. Labels:
+    ![labels](images/labels.png)
+    Every time you notice a bug, add it to the issues with the label “bug”. Whenever you/team decide to add a new features, enhancements can be used. Maybe there is a bug in the library you use which is out of your capability to fix…”wontfix”. You can also assign people to issues so that teammates don’t step into each other’s work. You also get an idea about what a teammate is working on. Seriously, you can use these labels anyway to your and your team’s productivity.
+
+    ..2. Milestones:
+    ![milestones](images/milestones.png)
+    Use this to properly scope your project and to make sure that your team is on track.
+
+#### 4. Pull Requests:
+    ..1. An extra checkpoint in merging a branch to master (or other branches)
+    ..2. Peers can look at the changes a merge will add and get a chance to review/comment on/ask question about the code
+    ..3. Peers can also checkout the branch to test it out
+
+#### 5. .gitignore
+    When you start collaborating on Android projects, you will start noticing how certain files from your Android project mess things up in other people's computers. There are some files that you never want to push to the repo. It can be hard to manually add certain files while keeping out others. .gitignore helps us here by never tracking any changes in that file. Usually, .gitignore contains certain extensions but it can also contain folders and specific filenames.
+
+    Here is a .gitignore you can use for your Android projects:
+
+    ```
+    # Built application files
+    *.apk
+    *.ap_
+
+    # Files for the ART/Dalvik VM
+    *.dex
+
+    # Java class files
+    *.class
+
+    # Generated files
+    bin/
+    gen/
+    out/
+
+    # Gradle files
+    .gradle/
+    build/
+
+    # Local configuration file (sdk path, etc)
+    local.properties
+
+    # Proguard folder generated by Eclipse
+    proguard/
+
+    # Log Files
+    *.log
+
+    # Android Studio Navigation editor temp files
+    .navigation/
+
+    # Android Studio captures folder
+    captures/
+
+    # Intellij
+    *.iml
+    .idea/workspace.xml
+
+    # Keystore files
+    *.jks
```

Comment:
I guess you're right. Damn AS, get on your shit.

On Thu, Jun 16, 2016 at 10:45 PM Shruti Iyer <notifications@github.com>
wrote:

> In lesson 1/README.md
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/4#discussion_r67454725>
> :
>
> > +
> > +    # Log Files
> > +    *.log
> > +
> > +    # Android Studio Navigation editor temp files
> > +    .navigation/
> > +
> > +    # Android Studio captures folder
> > +    captures/
> > +
> > +    # Intellij
> > +    *.iml
> > +    .idea/workspace.xml
> > +
> > +    # Keystore files
> > +    *.jks
>
> I haven't seen it do it. I had to add gitignores to all the projects I
> have done. Also, it is just good to talk about gitignores
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly, view it on GitHub
> <https://github.com/MobileProto16/Mobile-Proto-16/pull/4/files/e44f3412f166fa880ea843c0809ab90170fbd394..b4bf2fe1541d0b69873ea80f4475b9812712fc04#r67454725>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AIhRRdVIq7-4PQFFpeGJ6JVD7rav2sA6ks5qMgpjgaJpZM4I3-pv>
> .
>